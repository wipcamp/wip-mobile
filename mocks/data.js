export const newProblem = {
    topic: "Test new problem",
    problem_type_id: 1,
    description: "Test new problem",
    report_id: 1
}

export const problem = [
    {
        id: 1,
        topic: "Test Problem 1",
        description: "Test Description 1",
        problem_type_id: 1,
        report_id: 1,
        is_solve: false,
        not_solve: false,
        created_at: "2018-02-02 17:50:01",
        updated_at: "2018-02-02 17:50:01"
    },
    {
        id: 2,
        topic: "Test Problem 2",
        description: "Test Description 2",
        problem_type_id: 2,
        report_id: 2,
        is_solve: false,
        not_solve: false,
        created_at: "2018-02-02 17:50:02",
        updated_at: "2018-02-02 17:50:02"
    }
]

export const categoryProblem = [
    {
        value: 1,
        label: "Test 1"
    },
    {
        value: 2,
        label: "Test 2"
    }
]

export const filter = 0

export const filterView = [
    {
        value: 0,
        label: "ทั้งหมด"
    },
    {
        label: 1,
        value: "Test 1"
    },
    {
        label: 2,
        value: "Test 2"
    }
]