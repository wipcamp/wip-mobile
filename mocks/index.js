const mockComponent = require.requireActual('react-native/jest/mockComponent')

module.exports = {
    initAll: () => {
      jest.mock('ScrollView', () => {
        const RealScrollView = require.requireActual('ScrollView')
        const ScrollView = mockComponent('ScrollView')
        ScrollView.prototype.render = RealScrollView.prototype.render
        return ScrollView
      })

      jest.mock('Dimensions', () => {
        const Dimensions = {
          get: jest.fn().mockReturnValue({width: 100, height:100})
        }
        return Dimensions
      })
    }
  }