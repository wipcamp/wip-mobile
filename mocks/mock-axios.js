import axios from 'axios'
import mockAxios from 'axios-mock-adapter'

import env from '../config'

const mock = new mockAxios(axios)

export default mock