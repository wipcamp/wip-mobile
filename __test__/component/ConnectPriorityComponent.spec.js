/**
 * @jest-environment node
 */

import React from 'react'
import renderer from 'react-test-renderer'

import mockStore from '../../mocks/redux-mock-store'
import { priority } from '../../mocks/data'

import ConnectPriority from '../../components/ConnectPriorityComponent'

it('rendering', () => {
    const tree = renderer
        .create(
            <ConnectPriority
                store={
                    mockStore({
                        PriorityReducer: { priority }
                    })
                }
            />
        )
        .toJSON()
    expect(tree).toMatchSnapshot()
})