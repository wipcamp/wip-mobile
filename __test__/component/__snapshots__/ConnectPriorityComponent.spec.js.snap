// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "paddingLeft": 10,
        "paddingRight": 10,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={
        Object {
          "bottom": 6,
          "left": 4,
          "right": 4,
          "top": 6,
        }
      }
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="box-only"
      style={undefined}
      testID={undefined}
    >
      <View
        onResponderRelease={[Function]}
        onStartShouldSetResponder={[Function]}
        pointerEvents="none"
        style={undefined}
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderBottomColor": "rgba(0, 0, 0, 0.38)",
              "borderBottomWidth": NaN,
              "height": 52,
              "paddingBottom": 5,
              "paddingTop": 20,
            }
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "position": "absolute",
                "top": 24.5,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              collapsable={undefined}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.38)",
                  "fontSize": 18,
                }
              }
            >
              Priority
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              autoCapitalize="sentences"
              data={
                Array [
                  Object {
                    "label": "hight",
                    "value": 1,
                  },
                  Object {
                    "label": "medium",
                    "value": 2,
                  },
                ]
              }
              disableFullscreenUI={true}
              editable={false}
              itemColor="rgba(0, 0, 0, .54)"
              itemCount={4}
              itemPadding={8}
              onBlur={[Function]}
              onChange={[Function]}
              onChangeText={[Function]}
              onContentSizeChange={[Function]}
              onFocus={[Function]}
              renderAccessory={[Function]}
              selectionColor="rgb(0, 145, 234)"
              shadeOpacity={0.12}
              style={
                Array [
                  Object {
                    "flex": 1,
                    "margin": 0,
                    "padding": 0,
                    "top": 2,
                  },
                  Object {
                    "color": "rgba(0, 0, 0, .87)",
                    "fontSize": 18,
                    "height": 27,
                  },
                  undefined,
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              style={
                Object {
                  "alignSelf": "flex-start",
                  "justifyContent": "center",
                  "top": 2,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 24,
                    "justifyContent": "center",
                    "width": 24,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "height": 6,
                      "overflow": "hidden",
                      "width": 12,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "height": 8,
                          "transform": Array [
                            Object {
                              "translateY": -4,
                            },
                            Object {
                              "rotate": "45deg",
                            },
                          ],
                          "width": 8,
                        },
                        Object {
                          "backgroundColor": "rgba(0, 0, 0, .38)",
                        },
                      ]
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          collapsable={undefined}
          style={
            Object {
              "flexDirection": "row",
              "height": 8,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "paddingVertical": 4,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                collapsable={undefined}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "rgb(213, 0, 0)",
                    "fontSize": 0,
                    "opacity": 0,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "paddingVertical": 4,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                collapsable={undefined}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "rgba(0, 0, 0, .38)",
                    "fontSize": 12,
                    "opacity": 1,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        pointerEvents="box-only"
        rippleCentered={false}
        rippleColor="rgba(0, 0, 0, .38)"
        rippleDuration={450}
        rippleFades={true}
        rippleOpacity={0.54}
        rippleSequential={true}
        rippleSize={0}
        style={
          Object {
            "height": 51,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 16,
          }
        }
        testID={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "bottom": 0,
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "borderRadius": 0,
              },
            ]
          }
        />
      </View>
    </View>
    <Modal
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={true}
      visible={false}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 100,
            "width": 100,
          }
        }
        testID={undefined}
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "rgb(255, 255, 255)",
              "borderRadius": 2,
              "height": 102,
              "left": undefined,
              "opacity": 0,
              "position": "absolute",
              "shadowColor": "rgba(0, 0, 0, 1.0)",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.54,
              "shadowRadius": 2,
              "top": undefined,
              "transform": Array [
                Object {
                  "translateY": -51,
                },
              ],
              "width": undefined,
            }
          }
        >
          <RCTScrollView
            DEPRECATED_sendUpdatedChildFrames={false}
            alwaysBounceHorizontal={undefined}
            alwaysBounceVertical={true}
            contentContainerStyle={
              Object {
                "paddingVertical": 8,
              }
            }
            onContentSizeChange={null}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={undefined}
            onResponderTerminationRequest={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            onScrollShouldSetResponder={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            scrollEnabled={false}
            scrollEventThrottle={undefined}
            sendMomentumEvents={false}
            style={
              Array [
                Object {
                  "flexDirection": "column",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "overflow": "scroll",
                },
                Object {
                  "borderRadius": 2,
                  "flex": 1,
                },
              ]
            }
          >
            <RCTScrollContentView
              collapsable={false}
              removeClippedSubviews={undefined}
              style={
                Array [
                  undefined,
                  Object {
                    "paddingVertical": 8,
                  },
                ]
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                focusAnimation={null}
                focusAnimationDuration={225}
                fontSize={18}
                hitSlop={
                  Object {
                    "bottom": 6,
                    "left": 4,
                    "right": 4,
                    "top": 6,
                  }
                }
                index={0}
                nativeID={undefined}
                onLayout={[Function]}
                onPress={[Function]}
                onPressIn={[Function]}
                onPressOut={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                pointerEvents="box-only"
                rippleCentered={false}
                rippleColor="rgba(0, 0, 0, .38)"
                rippleDuration={450}
                rippleFades={true}
                rippleOpacity={0.54}
                rippleSequential={true}
                rippleSize={0}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 0,
                    "height": 43,
                    "justifyContent": "center",
                    "paddingLeft": undefined,
                    "paddingRight": undefined,
                    "paddingVertical": 8,
                  }
                }
                testID={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      undefined,
                      Object {
                        "color": "rgba(0, 0, 0, .87)",
                        "fontSize": 18,
                      },
                    ]
                  }
                >
                  hight
                </Text>
                <View
                  collapsable={undefined}
                  style={
                    Object {
                      "backgroundColor": "rgba(0, 0, 0, .38)",
                      "borderRadius": 0,
                      "bottom": 0,
                      "left": 0,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "bottom": 0,
                        "left": 0,
                        "overflow": "hidden",
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "borderRadius": 0,
                      },
                    ]
                  }
                />
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                focusAnimation={null}
                focusAnimationDuration={225}
                fontSize={18}
                hitSlop={
                  Object {
                    "bottom": 6,
                    "left": 4,
                    "right": 4,
                    "top": 6,
                  }
                }
                index={1}
                nativeID={undefined}
                onLayout={[Function]}
                onPress={[Function]}
                onPressIn={[Function]}
                onPressOut={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                pointerEvents="box-only"
                rippleCentered={false}
                rippleColor="rgba(0, 0, 0, .38)"
                rippleDuration={450}
                rippleFades={true}
                rippleOpacity={0.54}
                rippleSequential={true}
                rippleSize={0}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 0,
                    "height": 43,
                    "justifyContent": "center",
                    "paddingLeft": undefined,
                    "paddingRight": undefined,
                    "paddingVertical": 8,
                  }
                }
                testID={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      undefined,
                      Object {
                        "color": "rgba(0, 0, 0, .87)",
                        "fontSize": 18,
                      },
                    ]
                  }
                >
                  medium
                </Text>
                <View
                  collapsable={undefined}
                  style={
                    Object {
                      "backgroundColor": "rgba(0, 0, 0, .38)",
                      "borderRadius": 0,
                      "bottom": 0,
                      "left": 0,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "bottom": 0,
                        "left": 0,
                        "overflow": "hidden",
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "borderRadius": 0,
                      },
                    ]
                  }
                />
              </View>
            </RCTScrollContentView>
          </RCTScrollView>
        </View>
      </View>
    </Modal>
  </View>
</View>
`;
